src/compiler/types.ts

```diff
@@ -5427,52 +5427,60 @@ export const enum ContextFlags {
     SkipBindingPatterns = 1 << 3, // Ignore contextual types applied by binding patterns
 }
 
+export type NodeBuilderFlags = bigint;
+
 // NOTE: If modifying this enum, must modify `TypeFormatFlags` too!
 // dprint-ignore
-export const enum NodeBuilderFlags {
-    None                                    = 0,
+export const NodeBuilderFlags = {

```

Comment:
This is most definitely a breaking change; it will cause anyone using the reverse mapping to break (including our own debugging helpers), and also not be inlinable by our build. Not sure I like this approach...

Reply:
I agree, this is a breaking change, but I feel like any refactor on NodeBuilderFlags is going to be a breaking change.

Could you point me to the debugging helpers? I want to see how is the enum used.

The other approaches I'm considering for refactor are:
- Divide the flags into multiple enums/Add a new enum. (i.e. `NodeBuilderOptionFlags`, `NodeBuilderErrorFlags`, `NodeBuilderStateFlags`). This keeps the reverse mapping but requires more changes on the typescript repo. Plus, I can sense there's going to be some issues when combining flags from different enums.
- Replace the enum with an options interface. Using `&` and `|` would be gone but instead we can define the options. It will also have the same limitation as using BigInt:
 ```js
const NodeBuilderFlags = {
    NoTruncation: true,
    SuppressAnyReturnType: true
    ...
}
```

Do you have any suggestion of preferred approach?

Reply:
We could have two enums, then convert existing APIs that take those flags to instead take an object with the two enums as props (leaving old callers still valid with the old values, while new callers who want new flags have to pass the object in).

Reply:
e.g. more or less like the createSourceFile options.

Reply:
Changed as suggested. I decided to use the `flags` on the new interface and move the `AllowUnresolvedNames` as a property instead. The idea being using properties as options instead of starting to populate another `enum`.

Let me know if you prefer this change.